src/azul/service/manifest_service.py

```diff
@@ -703,10 +704,55 @@ def write_to(self, output: IO[str]) -> Optional[str]:
             url = furl(config.service_endpoint(),
                        path=f'/repository/files/{uuid}',
                        args=dict(version=version, catalog=self.catalog))
+            name = self._validate_path(name)
             output.write(f'url={self._option(url.url)}\n'
                          f'output={self._option(name)}\n\n')
         return None
 
+    @classmethod
+    def _validate_path(cls, path: str) -> str:
+        """
+        >>> f = CurlManifestGenerator._validate_path
+        >>> f('foo/bar/\\x1F/file')
+        Traceback (most recent call last):
+        ...
+        azul.RequirementError: Invalid file name 'foo/bar/\\x1f/file'
+
+        >>> f('foo/bar/\\/file')
+        Traceback (most recent call last):
+        ...
+        azul.RequirementError: Invalid file name 'foo/bar/\\\\/file'
+
+        >>> f('foo/bar/COM6/file')
+        Traceback (most recent call last):
+        ...
+        azul.RequirementError: Invalid file name 'foo/bar/COM6/file'
+
+        >>> f('foo/bar/ / baz/file')
+        Traceback (most recent call last):
+        ...
+        azul.RequirementError: Invalid file name 'foo/bar/ / baz/file'
+
+        >>> f('foo/bar?/fi*le|')
+        'foo/bar_/fi_le_'
+
+        >>> f('foo/bar/file.fastqgz')
+        'foo/bar/file.fastqgz'
+        """
+        msg = f'Invalid file name {path!r}'
+        for character in path:
+            reject(character == u'\\' or unicodedata.category(character) == 'Cc', msg)
```

Comment:
What's this for?

Reply:
its probably slow, too

Reply:
unicode literals are deprected, all strings are unicode in python now

Reply:
It looks like [`unicodedata`](https://docs.python.org/3/library/unicodedata.html#unicodedata.category) - "This module provides access to the Unicode Character Database (UCD) which defines character properties for all Unicode characters."
The `category` method returns `Cc` if the unicode character is a command. As you instructed in the [ticket](https://github.com/DataBiosphere/azul/issues/2398#issue-733576416) - "1. Fail if characters \x00 to \x1F or \ (backslash) occur in file paths. …". From the package documentation, I would assume each usage is a data base lookup. 
I thought it would be more robust to use this package, but do you have something else in mind? 
The first time around I generated the ASCII characters in a list comprehension by iterating over 0-f and 0-1, would that be better?

Reply:
> As you instructed 

Do my instructions mention Unicode or the control character category?

> From the package documentation, I would assume

Don't assume. Python is open source, so you can always check that.

> I thought it would be more robust to use this package,

More robust than using what?

> do you have something else in mind?

Yes.

> The first time around I generated the ASCII characters in a list comprehension by iterating over 0-f and 0-1, would that be better?

Probably not. But I can't be sure because I don't understand what you mean by "0-f and 0-1". That's not a notation I am familiar with.

As a learning experience I want you to figure this out on your own. There is a very simple solution that translates directly from my specification.

Reply:
>Do my instructions mention Unicode or the control character category?

They did not.

>More robust than using what?

Not sure 😅